staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -298,34 +326,26 @@ func (o *DebugOptions) visitPod(ctx context.Context, pod *corev1.Pod) (*corev1.P
 	return pod, debugContainer.Name, nil
 }
 
-func containerNames(pod *corev1.Pod) map[string]bool {
-	names := map[string]bool{}
-	for _, c := range pod.Spec.Containers {
-		names[c.Name] = true
-	}
-	for _, c := range pod.Spec.InitContainers {
-		names[c.Name] = true
+// debugByCopy runs a copy of the target Pod with a debug container added or an original container modified
+func (o *DebugOptions) debugByCopy(ctx context.Context, pod *corev1.Pod) (*corev1.Pod, string, error) {
+	copied, dc := o.generatePodCopyWithDebugContainer(pod)
+	if o.DeleteOld {
```

Comment:
If there's an error creating the new pod I would expect debug to return without deleting the old pod. I can see an argument for deleting the pod first, for example if you wanted to reschedule on a particular resource constrained node. I don't think this will be a common use case, though.

I lean towards create before delete. What do you think?

Reply:
I agree, I just wanted to make sure the old pod was deleted before the plugin proceeded to wait and attach and did not consider about the order of deletion and creation. Deletion after creation is a bette choice as you explained I think.

Reply:
Addressed in https://github.com/kubernetes/kubernetes/commit/b74f34a746fdef9c3c39c60df3b351e99e583025